java/ql/src/semmle/code/java/frameworks/spring/SpringUi.qll

```diff
@@ -0,0 +1,34 @@
+import java
+
+module SpringUi {
+  private class FlowSummaries extends SummaryModelCsv {
+    override predicate row(string row) {
+      row = [
+        "org.springframework.ui;Model;true;addAttribute;(String,Object);;Argument[0];MapKey of Argument[-1];value",
+        "org.springframework.ui;Model;true;addAttribute;(String,Object);;Argument[1];MapValue of Argument[-1];value",
+        "org.springframework.ui;Model;true;addAttribute;(Object);;Argument[0];MapValue of Argument[-1];value",
+        "org.springframework.ui;Model;true;addAllAttributes;(Collection);;Element of Argument[0];MapValue of Argument[-1];value",
+        "org.springframework.ui;Model;true;addAllAttributes;(Map<String,?>);;MapKey of Argument[0];MapKey of Argument[-1];value",
+        "org.springframework.ui;Model;true;addAllAttributes;(Map<String,?>);;MapValue of Argument[0];MapValue of Argument[-1];value",
+        "org.springframework.ui;Model;true;mergeAttributes;(Map<String,?>);;MapKey of Argument[0];MapKey of Argument[-1];value",
+        "org.springframework.ui;Model;true;mergeAttributes;(Map<String,?>);;MapValue of Argument[0];MapValue of Argument[-1];value",
+        "org.springframework.ui;Model;true;getAttribute;;;MapValue of Argument[-1];ReturnValue;value",
+        "org.springframework.ui;Model;true;mergeAttributes;;;MapValue of Argument[-1];ReturnValue;value",
+        "org.springframework.ui;Model;true;asMap;;;MapKey of Argument[-1];MapKey of ReturnValue;value",
+        "org.springframework.ui;Model;true;asMap;;;MapValue of Argument[-1];MapValue of ReturnValue;value",
+        "org.springframework.ui;ModelMap;false;ModelMap;(Object);;Argument[0];MapValue of ReturnValue;value",
+        "org.springframework.ui;ModelMap;false;ModelMap;(String, Object);;Argument[0];MapKey of ReturnValue;value",
+        "org.springframework.ui;ModelMap;false;ModelMap;(String, Object);;Argument[0];MapValue of ReturnValue;value",
```

Comment:
```suggestion
        "org.springframework.ui;ModelMap;false;ModelMap;(String,Object);;Argument[0];MapKey of ReturnValue;value",
        "org.springframework.ui;ModelMap;false;ModelMap;(String,Object);;Argument[0];MapValue of ReturnValue;value",
```

Last time I checked, spaces were causing the signature to not match. Do you know if this works correctly as it is?

Reply:
It doesn't, that's corrected in https://github.com/github/codeql/pull/6030